File: PROBLEMS/binary_search/new_year_and_hurry_fixed.py
def n_binary_search(low, high, k):
    best = 0  # Stores the maximum number of problems solved

    while low <= high:
        mid = low + (high - low) // 2  # Avoid overflow
        time_spent = sum(5 * i for i in range(1, mid + 1))  # Calculate total time for 'mid' problems

        if time_spent + k <= 240:  # Check if the total time is valid
            best = mid  # Update the best result
            low = mid + 1  # Try solving more problems
        else:
            high = mid - 1  # Reduce the number of problems

    return best  # Return the maximum number of problems possible


if __name__ == "__main__":
    n, k = map(int, input().split())
    print(n_binary_search(0, n, k))

File: PROBLEMS/binary_search/new_year_and_hurry_fixed.cpp
#include <bits/stdc++.h>
using namespace std;

// There are n problems
// The i-th problem takes 5*i minutes to be solved
// He needs k minutes to get to the event
// He will start the contest at 20:00 and need to be there until 00:00

int n_binary_search(int low, int high, int k) {
    int best = 0; // Armazena o maior número de problemas resolvidos

    while (low <= high) {
        int mid = low + (high - low) / 2; // Evitar overflow
        int time_spent = 0;

        for (int i = 1; i <= mid; i++) {
            time_spent += 5 * i;
        }

        if (time_spent + k <= 240) { // Verifica se o tempo total é válido
            best = mid; // Atualiza o melhor resultado
            low = mid + 1; // Tenta resolver mais problemas
        } else {
            high = mid - 1; // Reduz o número de problemas
        }
    }

    return best; // Retorna o maior número de problemas possíveis
}

int main() {
    int n, k;
    cin >> n >> k;
    cout << n_binary_search(0, n, k);
}
